### ISSUE TITLE
[bug] x86_64-apple-darwin11 native_cctools error on Ubuntu 20.04 and later

### ISSUE THREAD

**Issue by xanimo** [dogecoin/dogecoin#3130]
# Bug Report

**Current behavior:**
After downloading and unpacking the correct macOS SDK to the appropriate location, when invoking `make -C depends HOST=x86_64-apple-darwin11 NO_QT=1`, I encounter the following error with cctools on `Ubuntu Jammy 22.04`:

```
checking for x86_64-pc-linux-gnu-gcc... /mnt/source/repos/dogecoin/depends/work/build/x86_64-apple-darwin11/native_cctools/807d6fd1be5d2224872e381870c0a75387fe05e6-5af33fefe86/toolchain/bin/clang
checking whether the C compiler works... no
configure: error: in `/mnt/source/repos/dogecoin/depends/work/build/x86_64-apple-darwin11/native_cctools/807d6fd1be5d2224872e381870c0a75387fe05e6-5af33fefe86/cctools':
configure: error: C compiler cannot create executables
See `config.log' for more details
make: *** [funcs.mk:247: /mnt/source/repos/dogecoin/depends/work/build/x86_64-apple-darwin11/native_cctools/807d6fd1be5d2224872e381870c0a75387fe05e6-5af33fefe86/cctools/.stamp_configured] Error 77
make: Leaving directory '/mnt/source/repos/dogecoin/depends'
```

**Expected behavior**
To be able to run `make -C depends HOST=x86_64-apple-darwin11 NO_QT=1` on `Ubuntu Jammy 22.04` without errors in order to then compile core.

**Steps to reproduce:**
On a machine with `Ubuntu Jammy 22.04`, download this repo and checkout to `1.14.7-dev` and invoke: `make -C depends HOST=x86_64-apple-darwin11 NO_QT=1`

**Dogecoin Core version**
`1.14.7-dev`

**Comment by patricklodder** [comment]
Does the same issue happen on focal? (iirc it does and is one of the things that prevented a clean focal gitian environment)

**Comment by xanimo** [comment]
> Does the same issue happen on focal? (iirc it does and is one of the things that prevented a clean focal gitian environment)

It does indeed. I've been working on a fix for it on focal all day.

**Comment by xanimo** [comment]
It's sort of been like taking 1 step forward 2 steps back wherein I fix one thing and then encounter a new error...:smile: 

Anyway I'm currently stuck on a boost error but have yet to figure it out:
```
#12 466.1   "/dogecoin/depends/x86_64-apple-darwin11/native/bin/clang++" "-target" "x86_64-apple-darwin11" "-mmacosx-version-min=10.8" "--sysroot" "/dogecoin/depends/SDKs/MacOSX10.11.sdk" "-stdlib=libc++" -c -x c++ -fvisibility-inlines-hidden -std=c++11 -fvisibility=hidden     -I/dogecoin/depends/x86_64-apple-darwin11/include     -m64 -O3 -Wall -fvisibility=hidden -Wno-inline -Wextra -Wno-long-long -Wno-unused-parameter -Wno-variadic-macros -Wunused-function -DBOOST_ALL_NO_LIB=1 -DBOOST_ATOMIC_STATIC_LINK=1 -DBOOST_THREAD_BUILD_LIB=1 -DBOOST_THREAD_DONT_USE_CHRONO -DBOOST_THREAD_POSIX -DNDEBUG -I"." -o "bin.v2/libs/thread/build/clang-linux-8.0.0/release/link-static/target-os-darwin/threadapi-pthread/threading-multi/visibility-hidden/pthread/thread.o" "libs/thread/src/pthread/thread.cpp"
#12 466.1 
#12 466.1 In file included from libs/thread/src/pthread/thread.cpp:9:
#12 466.1 In file included from ./boost/thread/detail/config.hpp:11:
#12 466.1 In file included from ./boost/config.hpp:44:
#12 466.1 ./boost/config/detail/select_stdlib_config.hpp:18:12: fatal error: 'cstddef' file not found
#12 466.1 #  include <cstddef>
#12 466.1            ^~~~~~~~~
#12 466.1 1 error generated.
```

Going to attempt again from scratch in order to minimize the amount of cherry-picks but for reference purposes in case anyone else is interested I made a simple docker repo to test on focal without having to push every time a change is made as I'm on 22.04:
https://github.com/xanimo/docker-dogecoin :shrug:

**Comment by patricklodder** [comment]
I know your pain. Normally, I'd mark this as wont-fix because of the first line of the depends system readme:

> The depends system is maintained and tested using Ubuntu Bionic

I think jammy is going to cause some more issues than we need to solve https://github.com/dogecoin/dogecoin/issues/3095 so focusing on focal is probably the way to go here.

**Comment by xanimo** [comment]
> I know your pain. Normally, I'd mark this as wont-fix because of the first line of the depends system readme:
> 
> > The depends system is maintained and tested using Ubuntu Bionic
> 
> I think jammy is going to cause some more issues than we need to solve #3095 so focusing on focal is probably the way to go here.

Agreed :smile_cat: I'm currently shooting for a 20.04 fix to address #3095 and if achieved I'll most likely close this and the other bug issue I opened.

**Comment by patricklodder** [comment]
Fixed the title, now it's relevant again ❤️

**Comment by xanimo** [comment]
If it doesn't hurt I'm fine with that. I ended up solving most of the issues I was facing by installing gcc-9 haha. :+1:

**Comment by xanimo** [comment]
> downgrading a system compiler is not a good solution, I think. 

agreed, is only a 🩹 until we fix the actual issues 👍
